tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -101,21 +101,26 @@ def read(self, filename, binary_mode=False, size=None, offset=None):
             binary_mode: bool, read as binary if True, otherwise text
             size: int, number of bytes or characters to read, otherwise
                 read all the contents of the file from the offset
-            offset: int, offset into file to read from, otherwise read
+            offset: int, offset into file to read from, in bytes; otherwise read
```

Comment:
Unfortunately I don't think it works to use a byte offset if we're not in binary mode.

For example when we pass an encoding to `io.open()` what we get is a `TextIOBase` in which `seek()` and `tell()` operate in terms of characters rather than bytes:
https://docs.python.org/3/library/io.html#io.TextIOBase.tell

For consistency, I think if we're in binary mode we want size and offset to represent bytes, and if we're in text mode we want them both to represent characters.  Otherwise we might end up doing text reads that return partial values of multibyte characters, which basically defeats the point of having a text mode.

Reply:
Completely agree.  This is actually what the code was already doing, but the variable names and docstrings were wrong/confusing.

I've clarified those, particularly emphasizing that a) we don't care what `f.tell()` returns, so long as we can pass it back to `f.seek()`, and b) similarly at the `FileSystem.read(...)` level, the continuation tokens should be opaque.